File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        if(len(nums)<3):
            return max(nums)
        num=set(nums)
        new=list(num)
        new.sort()
        print(new)
        n=len(new)
        if(n==1):
            return new[0]
        if(n==2):
            return new[1]
        return new[len(new)-3]
        

File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        Set<Integer> set=new HashSet<>();
        for(int i=0;i<nums.length;i++)set.add(nums[i]);
        List<Integer> ls=new ArrayList<>();
        for(int i:set)ls.add(i);
        Collections.sort(ls);
        if(ls.size()<3) {
            if(ls.size()==2)return ls.get(1);
            return ls.get(0);
        }
        return ls.get(ls.size()-3);
    }
}